contracts/hardhat.config.ts

```diff
@@ -22,9 +22,15 @@ const config: HardhatUserConfig = {
   solidity: {
     version: "0.8.20",
     settings: {
+      viaIR: true,
```

Comment:
Some advanced way to compile in order to optimize the solidity code as I was running into issues. I think it's not necessary anymore since I removed the tx decoder anyway.